_safari/Save to Pocket Extension/Common/API.swift

```diff
@@ -118,30 +124,93 @@ class SaveToPocketAPI: SafariExtensionHandler{
         "account": "1",
         "grant_type": "extension"
       ]
-      
-      
-      let requestInfo: [String : Any] = ["url" : "https://getpocket.com/v3/oauth/authorize.php",
-                                         "method" : "POST",
-                                         "parameters" : requestData
+
+
+      let requestInfo: [String : Any] = [
+        "url" : "https://getpocket.com/v3/oauth/authorize.php",
+        "method" : "POST",
+        "parameters" : requestData
       ];
-      
-      NSLog("Auth Request: (\(String(describing: requestInfo)))")
+
+      //
+      NSLog("Request Auth: (\(String(describing: requestInfo)))")
       Utilities.request(from: page, userInfo: requestInfo) { result in
         switch result {
+
         case .failure(_):
           NSLog("Auth Failed: (\(String(describing: requestInfo)))")
+
         case .success(let data):
           guard let authJSON = try? JSONDecoder().decode(AuthResponse.self, from: data) else {
             NSLog("Auth Failed: (\(String(describing: requestInfo)))")
             completion(.failure(.auth))
             return
           }
+
           // Store Account data
           let defaults = UserDefaults.standard
           defaults.set(authJSON.access_token, forKey: "access_token")
+
           completion(.success(authJSON.access_token))
+
+        }
+      }
+    }
+  }
+
+  static func saveToPocket(
+    from page: SFSafariPage,
+    url: String,
+    access_token: String,
+    completion: @escaping (Result<Any, RequestError>) -> Void
+    ) -> Void {
+
+
+    // Build Action
+    let saveAction: [String : Any] = [
+      "action": "add",
+      "url": url
+    ]
+
+    // Build request data dictionary
+    let requestData: [String : Any] = [
+      "consumer_key": "70018-b83d4728573df682a7c50b3d",
+      "access_token": access_token,
+      "actions": [saveAction]
+    ]
+
+
+    let requestInfo: [String : Any] = [
+      "url" : "https://getpocket.com/v3/send/",
+      "method" : "POST",
+      "parameters" : requestData
+    ];
+
+    Utilities.request(from: page, userInfo: requestInfo) { result in
+      switch result {
+
+      case .failure(_):
+        NSLog("Save Failed: (\(String(describing: requestInfo)))")
```

Comment:
Should we `completion(.failure(error))` here and `return`?

Reply:
@collectedmind This is the only issue that stood out to me.

Reply:
👍